api/task_processor/models.py

```diff
@@ -131,6 +132,10 @@ class Meta:
 
     @property
     def should_execute(self) -> bool:
+        now = timezone.now()
+        if self.first_run_time and self.first_run_time > now.time():
```

Comment:
Should we add a comment here? It took more than ideal time to understand this, but when I finally did, it sort of became obvious. What do you think?

Reply:
I've added a comment and slightly tweaked the logic, as I realised that the logic that was there was susceptible to issues if the task start time slipped (over a long period of time) such that it drifted beyond midnight it would never run again.